database/sql/V048__create_settlement_rewards.sql

```diff
@@ -12,12 +15,12 @@ CREATE TABLE settlement_observations (
   -- table, read from the transaction receipt
   block_number bigint NOT NULL,
   log_index bigint NOT NULL,
-  gas_used numeric(78,0) NOT NULL,
-  effective_gas_price numeric(78,0) NOT NULL,
+  gas_used numeric(78,0),
+  effective_gas_price numeric(78,0),
```

Comment:
In my mind, these need to be nullable.

Specifically, the `driver`/`solver` needs to ask to populate this **before** mining the transaction, when `gas_used` and `effective_gas_price` is unknown.

Reply:
Agree they should be nullable, but I don't understand why it needs to be populated before mining the transaction? I insert it for the first time when the `settlement event` is raised - transaction mined.